File: samples/spring-mvc/src/main/java/io/openapiprocessor/samples/EchoController.java
/*
 * Copyright 2019 https://github.com/openapi-processor/openapi-processor-samples
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.EchoApi;
import org.springframework.web.bind.annotation.RestController;

/**
 * Implementation of the echo api defined in src/api/openapi.yaml.
 */
@RestController
public class EchoController implements EchoApi {

    @Override
    public String getEcho (String source) {
        return source;
    }

}

File: samples/spring-mvc-boot-4-packages-kt/src/main/kotlin/io/openapiprocessor/samples/echo/EchoController.kt
/*
 * Copyright 2015 https://github.com/openapi-processor/openapi-processor-samples
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiprocessor.samples.echo

import org.springframework.web.bind.annotation.RestController

@RestController
class EchoController: EchoApi {
    override fun getEcho(source: String): String {
        return source
    }
}
